timemachine/cpp/src/bound_potential.cu

```diff
@@ -58,12 +55,20 @@ void BoundPotential::execute_host(
     }
 };
 
+void BoundPotential::set_params(const std::vector<double> params) {
+    this->size = params.size();
+    if (this->size > 0) {
```

Comment:
This should call down to `set_params_device` for consistency with our other APIs and should have the same behavior of not allowing you to resize/reallocate the underlying parameters buffer (We removed this option in https://github.com/proteneer/timemachine/pull/1063)

Reply:
Simply so that calling `set_params_device` and `set_params` has identical behavior. Which is consistent with our other APIs that have a device and host method (https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/nonbonded_all_pairs.cu#L106-L114) that way it has exactly the same behavior called from device code or host code.

Reply:
OK. Consistent behavior for `set_params` and `set_params_device` makes sense to me. Was just confused by

> This should call down to set_params_device

Reply:
Changed the behavior to allocate a static buffer for the lifetime of the `BoundPotential` object and raise an exception if `set_params` is called with parameters larger than the buffer in https://github.com/proteneer/timemachine/pull/1119/commits/c0eaa8de677f2f60526fd25856b6d03dd85deeb9.